tamr_toolbox/data_io/core_connect.py

```diff
@@ -0,0 +1,89 @@
+"""Tasks related to interacting with the Tamr Core-connect"""
+import json
+import logging
+from tamr_toolbox.models.data_type import JsonDict
+from tamr_unify_client import Client
+from tamr_toolbox.utils import version
+import os
+
+os.environ.setdefault("TAMR_MIN_VERSION_W_CORE_CONNECT", "2022.005.0")
+
+LOGGER = logging.getLogger(__name__)
+
+
+def ingest_dataset(
+    *,
+    client: Client,
+    jdbc_connect: JsonDict,
+    dataset_name: str,
+    query: str,
+    primary_key=None,
+    truncate_tamr_dataset=None,
+    retrieve_connect_metadata=None,
+    retrieve_source_metadata=None,
+) -> JsonDict:
+    """
+    Ingest a dataset into Tamr via core_connect given query config, dataset name, query string,
+    and optional list of columns for primary key
+
+    Args:
+        client: Tamr unify client object.
+        jdbc_connect: queryConfig used by the API call.
+        dataset_name: Name of Tamr dataset.
+        query: jdbc query to execute in the database and results of which will be loaded into Tamr.
+        primary_key(optional): list of columns to use as primary key.
+        truncate_tamr_dataset(optional): Truncate the Tamr dataset if it exists.
+        retrieve_connect_metadata(optional): Core Connect retrieves the metadata for services.
+        retrieve_source_metadata(optional): Only applies to Snowflake.
+
+    Raises:
+        HTTPError: if the call to ingest the dataset was unsuccessful
+    """
+    # Check Tamr version
+    if version.is_version_condition_met(
+        tamr_version=version.current(client),
+        min_version=os.environ["TAMR_MIN_VERSION_W_CORE_CONNECT"],
+    ):
+        LOGGER.info(
+            f"Tamr version is equal to or after {os.environ['TAMR_MIN_VERSION_W_CORE_CONNECT']}"
+        )
+    else:
+        error_message = "Core-connect is not available in current version of Tamr."
+        LOGGER.error(error_message)
+        raise Exception(error_message)
+
+    # handle primary key
+    if primary_key is None:
+        primary_key = []
+    else:
+        primary_key = primary_key.split(",")
+
+    # ingest data
+    api_path = "/api/connect/jdbcIngest/ingest"
```

Comment:
Given that you are starting out with just the jdbc ingest endpoint, I have the following suggestion to you. Why not leverage the existing df-connect code? The code looks almost identical to the df-connect ingest function right? How about adding a version check in the df-connect module. If the version is >=v2022.005, use the core-connect endpoint. Else, use the df-connect endpoint. That way, we do not break functionality. I do envision needing a separate core-connect module as we add other cloud endpoints that are present in core-connect but not df-connect... But the jdbc ingest is common between both. I suggest we treat it as such.

Reply:
I agree that there will be more functions added in the core-connect module for other ingestion and exporting functions. I would prefer to keep the core-connect ingest function in core-connect module for a couple reasons: 1.  it is confusing to call a core-connect ingestion function from the df-connect module; 2. there are more options for core-connect ingestion function than df-connect; 3. it is easy to test the functions separately.

Reply:
1. function renamed to jdbc_ingest.
2. As we cannot test df-connect endpoint and core-connect endpoint in the same instance, we will leave df-connect and core-connect separate modules.

Reply:
Resolved internally - will not maintain backwards compatibility with df-connect.